diff --git a/BUILD.gn b/BUILD.gn
index 53053ea..9bcebdb 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -984,6 +984,7 @@ source_set("libcef_static") {
     "//third_party/blink/public:blink",
     "//third_party/brotli:dec",
     "//third_party/cld_3/src/src:cld_3",
+    "//third_party/highway:libhwy",
     "//third_party/crashpad/crashpad/handler",
     "//third_party/hunspell",
     "//third_party/leveldatabase",
diff --git a/include/cef_v8.h b/include/cef_v8.h
index be77ff3..bfa95ea 100644
--- a/include/cef_v8.h
+++ b/include/cef_v8.h
@@ -1027,6 +1027,16 @@ class CefV8Value : public virtual CefBaseRefCounted {
   ///
   /*--cef()--*/
   virtual bool RejectPromise(const CefString& errorMsg) = 0;
+
+#if CEF_API_ADDED(CEF_EXPERIMENTAL)
+  ///
+  /// Returns the filesystem path for a DOM File object. Returns an empty
+  /// string if this value is not a File object or if the File is not backed by
+  /// a path on disk.
+  ///
+  /*--cef(added=experimental)--*/
+  virtual CefString GetPathForFile() = 0;
+#endif
 };
 
 ///
diff --git a/libcef/renderer/v8_impl.cc b/libcef/renderer/v8_impl.cc
index 0bcf093..83d12c9 100644
--- a/libcef/renderer/v8_impl.cc
+++ b/libcef/renderer/v8_impl.cc
@@ -40,6 +40,8 @@
 #include "cef/libcef/renderer/v8_impl.h"
 #include "third_party/abseil-cpp/absl/base/attributes.h"
 #include "third_party/blink/public/web/blink.h"
+#include "third_party/blink/renderer/bindings/core/v8/v8_file.h"
+#include "third_party/blink/renderer/core/fileapi/file.h"
 #include "third_party/blink/public/web/web_frame.h"
 #include "third_party/blink/public/web/web_local_frame.h"
 #include "third_party/blink/public/web/web_script_controller.h"
@@ -2006,6 +2008,26 @@ CefString CefV8ValueImpl::GetStringValue() {
   return rv;
 }
 
+CefString CefV8ValueImpl::GetPathForFile() {
+  CefString rv;
+  CEF_V8_REQUIRE_MLT_RETURN(rv);
+  if (type_ != TYPE_OBJECT || !CEF_V8_HANDLE_IS_VALID()) {
+    return rv;
+  }
+
+  v8::HandleScope handle_scope(handle_->isolate());
+  auto value = handle_->GetNewV8Handle(false);
+  if (auto* file = blink::V8File::ToWrappable(handle_->isolate(), value);
+      file && file->HasBackingFile()) {
+    auto path = file->GetPath();
+    if (!path.empty()) {
+      path.Ensure16Bit();
+      rv = CefString(path.Span16().data(), path.length());
+    }
+  }
+  return rv;
+}
+
 bool CefV8ValueImpl::IsUserCreated() {
   CEF_V8_REQUIRE_OBJECT_RETURN(false);
 
diff --git a/libcef/renderer/v8_impl.h b/libcef/renderer/v8_impl.h
index 1c5c146..7c4530d 100644
--- a/libcef/renderer/v8_impl.h
+++ b/libcef/renderer/v8_impl.h
@@ -286,6 +286,7 @@ class CefV8ValueImpl : public CefV8Value {
 
   bool ResolvePromise(CefRefPtr<CefV8Value> arg) override;
   bool RejectPromise(const CefString& errorMsg) override;
+  CefString GetPathForFile() override;
 
  private:
   // Test for and record any exception.
